django_emarsys/config.py

```diff
@@ -0,0 +1,184 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+import re
+from collections import Counter
+from itertools import chain
+
+from django.apps import apps
+from django.conf import settings
+from django.core.checks import Critical, Error, Warning
+
+
+_VALID_ARG = re.compile(r'^[a-z][a-z0-9_]*$')
+
+
+def _flatten(l):
+    """
+    Flatten a list of lists into a list
+    """
+    return list(chain.from_iterable(l))
+
+
+def _find_duplicates(l):
+    """
+    Return non-unique elements in a list.
+
+    Should be in Python.
+    """
+    return [item for item, count in Counter(l).items() if count > 1]
+
+
+def validate_settings():
+    """
+    Turn settings into a configuration dict that `validate_configuration` can
+    check.
+
+    The extra `hasattr()` check is necessary to make this work with overridden
+    settings from tests. If a setting existed in the original settings object,
+    but is then deleted in overridden settings, then `dir()` will report it
+    present, but accessing it raises an `AttributeError`.
+
+    Minimal test for this:
+
+    >>> from django.conf import settings
+    >>> from django.test.utils import override_settings
+    >>> @override_settings()
+    >>> def test_strange_dir_issue():
+    >>>     delattr(settings, 'DEBUG')
+    >>>     assert 'DEBUG' in dir(settings)
+    >>>     assert not hasattr(settings, 'DEBUG')
+    >>>
+    >>> test_strange_dir_issue()
```

Comment:
The part about hasattr is an implementation comment, not documentation, so I'd put it behind `#` comments.